src/core/analytics.js

```diff
@@ -1130,35 +1223,13 @@ class Analytics {
     // logger.debug("inside load ");
     if (this.loaded) return;
 
-    // check if the below features are available in the browser or not
-    // If not present dynamically load from the polyfill cdn, unless
-    // the options are configured not to.
-    const polyfillIfRequired =
-      options && typeof options.polyfillIfRequired === 'boolean'
-        ? options.polyfillIfRequired
-        : true;
-    if (
-      polyfillIfRequired &&
-      (!String.prototype.endsWith ||
-        !String.prototype.startsWith ||
-        !String.prototype.includes ||
-        !Array.prototype.find ||
-        !Array.prototype.includes ||
-        !Promise ||
-        !Object.entries ||
-        !Object.values ||
-        !String.prototype.replaceAll ||
-        !this.isDatasetAvailable())
-    ) {
+    if (this.arePolyfillsRequired()) {
       const id = 'polyfill';
       ScriptLoader(id, POLYFILL_URL, { skipDatasetAttributes: true });
       const self = this;
       const interval = setInterval(() => {
         // check if the polyfill is loaded
-        // In chrome 83 and below versions ID of a script is not part of window's scope
-        // even though it is loaded and returns false for <window.hasOwnProperty("polyfill")> this.
-        // So, added another checking to fulfill that purpose.
-        if (window.hasOwnProperty(id) || document.getElementById(id) !== null) {
+        if (window.hasOwnProperty(id) || !self.arePolyfillsRequired()) {
```

Comment:
why we needed to change this one?

Reply:
This change was made in the original DMT branch. The reason is, in some browser versions both of the below checking didn't work:
- window.hasOwnProperty(id)
- document.getElementById(id) !== null

Can't remember the exact browsers, this was done a long ago.

Reply:
document.getElementById is always returning HTMLElement | null from very old times so I am surpised. If I had to delete somethig here it would be window.hasOwnProperty(id) instead of the document one.

How did we test this one?

Reply:
Reverted the changes as per discussion